src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs

```diff
@@ -0,0 +1,445 @@
+}
+
+class Test2 : I1 {}
+";
+            var compilation1 = CreateCompilationWithMscorlib(source1, options: TestOptions.DebugDll,
+                                                             parseOptions: TestOptions.Regular.WithLanguageVersion(LanguageVersion.Latest));
+            Assert.True(compilation1.Assembly.RuntimeSupportsDefaultInterfaceImplementation);
+            compilation1.VerifyDiagnostics(
+                // (7,15): error CS0736: 'Test1' does not implement interface member 'I1.M1()'. 'Test1.M1()' cannot implement an interface member because it is static.
+                // class Test1 : I1
+                Diagnostic(ErrorCode.ERR_CloseUnimplementedInterfaceMemberStatic, "I1").WithArguments("Test1", "I1.M1()", "Test1.M1()").WithLocation(7, 15)
+                );
+        }
+
+        [Fact]
+        public void MethodImplementation_04()
+        {
+            var source1 =
+@"
+interface I1
+{
+    void M1() {}
+    int M2() => 1; 
+}
+
+class Test1 : I1
+{
+    public int M1() { return 0; }
+    public ref int M2() { throw null; }
+}
+
+class Test2 : I1 {}
+";
+            var compilation1 = CreateCompilationWithMscorlib(source1, options: TestOptions.DebugDll,
+                                                             parseOptions: TestOptions.Regular.WithLanguageVersion(LanguageVersion.Latest));
+            Assert.True(compilation1.Assembly.RuntimeSupportsDefaultInterfaceImplementation);
+            compilation1.VerifyDiagnostics(
+                // (8,15): error CS8152: 'Test1' does not implement interface member 'I1.M2()'. 'Test1.M2()' cannot implement 'I1.M2()' because it does not return by value
+                // class Test1 : I1
+                Diagnostic(ErrorCode.ERR_CloseUnimplementedInterfaceMemberWrongRefReturn, "I1").WithArguments("Test1", "I1.M2()", "Test1.M2()", "value").WithLocation(8, 15),
+                // (8,15): error CS0738: 'Test1' does not implement interface member 'I1.M1()'. 'Test1.M1()' cannot implement 'I1.M1()' because it does not have the matching return type of 'void'.
+                // class Test1 : I1
+                Diagnostic(ErrorCode.ERR_CloseUnimplementedInterfaceMemberWrongReturnType, "I1").WithArguments("Test1", "I1.M1()", "Test1.M1()", "void").WithLocation(8, 15)
+                );
+        }
+
+        private static MetadataReference MscorlibRefWithoutSharingCachedSymbols
+        {
+            get
+            {
+                return ((AssemblyMetadata)((MetadataImageReference)MscorlibRef).GetMetadata()).CopyWithoutSharingCachedSymbols().
+                    GetReference(display: "mscorlib.v4_0_30319.dll");
+            }
+        }
+
+        [Fact]
+        public void MethodImplementation_05()
+        {
+            var source1 =
+@"
+public interface I1
+{
+    void M1() 
+    {
+        System.Console.WriteLine(""M1"");
+    }
+}
+
+class Test1 : I1
+{}
+";
+            var mscorLibRef = MscorlibRefWithoutSharingCachedSymbols;
```

Comment:
>MscorlibRefWithoutSharingCachedSymbols [](start = 30, length = 38)

Are you avoiding sharing so as to have an independent copy of `RuntimeSupportsDefaultInterfaceImplementation`? If so a comment somewhere would be helpful.